Project/Modules/Controllers/PostsController.cs

```diff
@@ -0,0 +1,73 @@
+﻿using blog_net_core.Project.Modules.Model.Entities;
+using blog_net_core.Project.Modules.Model.Repositories;
+using blog_net_core.Project.Modules.Services;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace blog_net_core.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class PostsController : ControllerBase
+    {
+        private readonly IPostService _postService;
+
+        public PostsController(IPostService postService)
+        {
+            _postService = postService;
+        }
+
+        [Route("")]
+        [HttpGet]
+        public IActionResult GetAll()
```

Comment:
Transform this function as an asynchronous function: 
clue: the return type  change this IActionResult type by Task<List<Post>>
*apply this change to all the functions.

Reply:
transformed to asynchronous task